components/mailgun/actions/suppress-email/suppress-email.js

```diff
@@ -0,0 +1,84 @@
+const mailgun = require("../../mailgun.app.js");
+
+module.exports = {
+  key: "mailgun-suppress-email",
+  name: "Mailgun Suppress Email",
+  description: "Add email to the Mailgun suppression list.",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    mailgun,
+    domain: {
+      propDefinition: [
+        mailgun,
+        "domain",
+      ],
+    },
+    email: {
+      propDefinition: [
+        mailgun,
+        "email",
+      ],
+    },
+    category: {
+      type: "string",
+      options: [
+        "bounces",
+        "unsubscribes",
+        "complaints",
+      ],
+    },
+    bounceErrorCode: {
+      type: "string",
+      label: "Bounce Error Code",
+      default: "550",
+      optional: true,
+    },
+    bounceErrorMessage: {
+      type: "string",
+      label: "Bounce Error Message",
+      optional: true,
+    },
+    unsubscribeFrom: {
+      type: "string",
+      label: "Tag to unsubscribe from",
+      description: "Use * to unsubscribe an address from all domain’s correspondence",
+      default: "*",
+      optional: true,
+    },
+    haltOnError: {
+      propDefinition: [
+        mailgun,
+        "haltOnError",
+      ],
+    },
+  },
+  async run () {
+    try {
+      const suppression = {
+        address: this.email,
+      };
+
+      switch (this.category) {
```

Comment:
Is a `default` case necessary to capture any type of code / error / tag? Or does the `suppression` object above capture the correct default case?

Reply:
@dylburger no default case needed, bounces and unsubscribes are the only cases that accept additional information, and all cases require address.